src/v0/sources/shopify_v2/trackEventsLayer.js

```diff
@@ -0,0 +1,274 @@
+  // async checkAndUpdateCartItems(inputEvent, redisData, metricMetadata) {
+  //   const cartToken = inputEvent.token || inputEvent.id;
+  //   const itemsHash = redisData?.itemsHash;
+  //   if (itemsHash) {
+  //     const newCartItemsHash = getHashLineItems(inputEvent);
+  //     const isCartValid = this.isValidCartEvent(newCartItemsHash, itemsHash);
+  //     if (!isCartValid) {
+  //       return false;
+  //     }
+  //     await this.updateCartItemsInRedis(cartToken, newCartItemsHash, metricMetadata);
+  //   } else {
+  //     const { created_at, updated_at } = inputEvent;
+  //     const timeDifference = Date.parse(updated_at) - Date.parse(created_at);
+  //     const isTimeWithinThreshold = timeDifference < maxTimeToIdentifyRSGeneratedCall;
+  //     const isLineItemsEmpty = inputEvent?.line_items?.length === 0;
+
+  //     if (isTimeWithinThreshold && isLineItemsEmpty) {
+  //       return false;
+  //     }
+  //   }
+  //   return true;
+  // },
+
+  /**
+   * This function will update the cart state in redis
+   * @param {*} updatedCartState
+   * @param {*} cart_token
+   */
+  // async updateCartState(updatedCartState, cart_token) {
+  //   await RedisDB.setVal(`${cart_token}`, updatedCartState);
+  // },
+  /**
+   * This function return the updated event name for carts_update event based on previous cart state
+   * And updates the state of cart in redis as well
+   * @param {*} event
+   * @param {*} redisData
+   */
+  // checkForProductAddedOrRemovedAndUpdate(event, redisData) {
+  //   const { productsListInfo } = redisData;
+  //   /*
+  //   productsListInfo = {
+  //     `productId+variantId` : quantity
+  //   }
+  //   */
+  //   let productsListInfoFromInput;
+  //   event?.line_items.forEach(product => {
+  //     const key = `${product.productId} + ${product.variantId}`;
+  //     const valFromPayload = product.quantity;
+  //     const prevVal = productsListInfo?.[key];
+  //     if (prevVal > valFromPayload) { }
+  //   })
+  // },
+
+  /**
+   * This function handles the event from shopify side which is mapped to rudder ecom event based upon the contents of payload.
+   * @param {*} event
+   * @param {*} eventName
+   * @param {*} redisData
+   * @returns the updated name of the payload
+   */
+  getUpdatedEventName(event, eventName, redisData) {
+    let updatedEventName;
+
+    // if (eventName === 'carts_update') {
+    //   return NO_OPERATION_SUCCESS
+    //   // this.checkForProductAddedOrRemovedAndUpdate(event, redisData);
+    //   // return 'carts_update';
+    // }
+    /* This function will check for cart_update if its is due Product Added or Product Removed and
+      for checkout_update which step is completed or started
+    */
+
+    if (eventName === 'checkouts_update') {
+      updatedEventName = 'checkout_step_viewed';
+      if (event.completed_at) {
+        updatedEventName = 'checkout_step_completed';
+      } else if (!event.gateway) {
+        updatedEventName = 'payment_info_entered';
```

Comment:
Can we validate if this event would be fired? and how it differs from the other checkout step events

Reply:
@Gauravudia has already validated it through extensive researching

Reply:
checkouts_update webhook from shopify is not a RudderStack standard ecommerce event. Converting it to RudderStack standard ecommerce events. Which will be used by downstream destinations.